contracts/BaseStrategy.sol

```diff
@@ -225,6 +233,16 @@ abstract contract BaseStrategy {
         require(msg.sender == keeper || msg.sender == strategist || msg.sender == governance(), "!authorized");
         _;
     }
+    modifier useChi() {
+        if (address(chiToken) == address(0)) {
+            _;
+            return;
+        }
+        uint256 gasStart = gasleft();
+        _;
+        uint256 gasSpent = 21000 + gasStart - gasleft() + 16 * msg.data.length;
+        chiToken.freeFromUpTo(msg.sender, (gasSpent + 14154) / 41130);
```

Comment:
comment on some of these literal values

Reply:
ok will add.

Reply:
I can't find proper explanation from the devs about the literal values.

The formula is from https://1inch-exchange.medium.com/1inch-introduces-chi-gastoken-d0bd5bb0f92b

Reply:
the last value in above link is `41947` ?
```
chi.freeFromUpTo(msg.sender, (gasSpent + 14154) / 41947);
```

Reply:
So 41947 would be the amout of gas saved by each CHI burnt.
14154 my guess is the gas used by freeFromUpTo.

Reply:
make 100% sense to me man. so it is a heuristic magic value then

Reply:
```suggestion
        // NOTE: Per EIP-2028, gas cost is 16 per (non-empty) byte in calldata
        uint256 gasSpent = 21000 + gasStart - gasleft() + 16 * msg.data.length;
        // NOTE: 41947 is the estimated amt of gas refund realized per CHI redeemed
        // NOTE: 14154 is the estimated cost of the call to `freeFromUpTo`
        chiToken.freeFromUpTo(msg.sender, (gasSpent + 14154) / 41130);
```